pkg/runner/agent.go

```diff
@@ -87,7 +88,7 @@ func (a *agentLoop) _saveEmptyLogs(ctx context.Context, environmentId string, ex
 	if execution.Workflow != nil {
 		workflowName = execution.Workflow.Name
 	}
-	return a.client.SaveExecutionLogs(ctx, environmentId, execution.Id, workflowName, nil)
+	return a.client.SaveExecutionLogs(ctx, environmentId, execution.Id, workflowName, strings.NewReader(""))
```

Comment:
This would go all the way to pkg/bufferedstream/bufferedstream.go:33, which then leads to a panic in stdlib /opt/homebrew/opt/go/libexec/src/io/io.go:429: `nr, er := src.Read(buf)`.

Reply:
This rather should be specifically `nil` - otherwise the behavior may be ambiguous. The problem is Golang's `http` implementation - if the `nil` or empty `bytes.Buffer` is passed - it's correctly assigning `Content-Length`, otherwise it may not assign `Content-Length` leading to errors on proxy/S3.

Reply:
Nevermind, it's passed through the `BufferedStream` anyway 👍